packages/clerk-js/src/utils/url.ts

```diff
@@ -103,24 +106,44 @@ export function buildURL(params: BuildURLParams, options: BuildURLOptions<boolea
   // Treat that hash part of the main URL as if it's another URL with a pathname and a search.
   // Another nested hash inside the top level hash (e.g. #my-hash#my-second-hash) is currently
   // not supported as there is no use case for it yet.
-  if (hashPath || hashSearch) {
+  if (hashPath || hashSearch || hashSearchParams) {
     // Parse the hash to a URL object
     const dummyUrlForHash = new URL(DUMMY_URL_BASE + url.hash.substring(1));
 
     // Join the current hash path and with the provided one
     dummyUrlForHash.pathname = joinPaths(dummyUrlForHash.pathname, hashPath || '');
 
-    // Merge search params
-    const hashSearchParams = getQueryParams(hashSearch || '');
-    for (const [key, val] of Object.entries(hashSearchParams)) {
+    // Merge search params from hashSearch string
+    const searchParamsFromHashSearchString = getQueryParams(hashSearch || '');
+    for (const [key, val] of Object.entries(searchParamsFromHashSearchString)) {
       dummyUrlForHash.searchParams.append(key, val as string);
     }
 
-    // Keep just the pathname and the search
+    // Merge search params from the hashSearchParams object
+    if (hashSearchParams) {
+      const paramsArr = Array.isArray(hashSearchParams) ? hashSearchParams : [hashSearchParams];
```

Comment:
Previously, the util supported passing hash params as a `string` only but the best way here is to simply pass a URLSearchParams object instead. I added the new functionality but kept the old one, will refactor all usages to the new format in a new PR